### PR TITLE
[CORE] Simplify code of offload scan

### PR DESCRIPTION
**Pull request by zml1206**
## What changes were proposed in this pull request?

1. Offload scan does not need to doValidate again.
2. Remove catch of FallbackByNativeValidation.

### CODE REVIEW

File: gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala [hunk 1]
```diff
@@ -118,21 +119,29 @@ object RasOffload {
...
+                    //
+                    // TODO: Tag the original plan with fallback reason. This is a non-trivial work
+                    //  in RAS as the query plan we got here may be a copy so may not propagate tags
+                    //  to original plan.
+                    from
+                  } else {
+                    offloadedPlan
+                  }
+                } catch {
+                  case _: GlutenNotSupportException =>
```
Comment (zml1206) [inline gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala:140]: Keep and disable RAS consistent.

Comment (zhztheplayer) [reply gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala:140]: Would you help check the corresponding logic in legacy planner to see whether we should only catch exceptions from `doValidate()` or catch more code like this way? Thanks!

**Review by zhztheplayer — COMMENTED** [review]
% one comment. Thanks.

File: gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala (hunk 1 above)
Comment (zml1206) [reply gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala:140]: I didn't find the place where GlutenNotSupportException is thrown in the offload rule. `doValidate()` contains `doValidateInternal()` and `doSchemaValidate()`, `doValidateInternal()` has been caught, `doSchemaValidate()` will not trigger an exception, so restore it？

Comment (zhztheplayer) [reply gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/enumerated/RasOffload.scala:140]: I see. So in this PR we can shrink the range of the catch block to only catch `doValidate()` to match legacy planner. Perhaps in future we should make all exceptions be caught in `doValidate` internally to make `doValidate` exception-free.

File: gluten-substrait/src/main/scala/org/apache/gluten/backendsapi/BackendSettingsApi.scala [hunk 2]
```diff
@@ -121,6 +121,8 @@ trait BackendSettingsApi {
 
   def supportCartesianProductExec(): Boolean = false
 
+  def supportCartesianProductExecWithCondition(): Boolean = true
```
Comment (zhztheplayer) [inline gluten-substrait/src/main/scala/org/apache/gluten/backendsapi/BackendSettingsApi.scala:124]: 👍

**Review by zhztheplayer — APPROVED** [review]
LGTM. Thank you very much for talking this code cleanup.

**Pull request merged**